Docs/SpellCheckingConfiguration.md

```diff
@@ -0,0 +1,46 @@
+# Spell checking configuration advice
```

Comment:
That's what the "When not to add entries to dictionary files" section is for. I wrote this documentation "organically" in the sense that I kept refining it based on the situations I encountered across all the projects we applied spell checking to.

I've added everything I think can help but do note that you also have to know about the configuration files to some extent, because addressing a false positive can be done in different ways. That's why this document says at the beginning that you need to be familiar with the configuration options first.

Can you give concrete examples where you wouldn't know what to do?

Reply:
Currently, you have to read and understand the full page to configure anything, it's too much for the everyday case (as opposed to e.g. when you're setting up a new repo). So, in the advice file I'd add something along the lines of:

> Is this a false positive and there's no actual spelling error? If the word is used only at most in a handful of cases, use [notes on ignore-line]. If it's used repeatedly all over the project, you'll need to extend the corresponding dictionary. For more info on how to configure spell-checking in-depth, check out [link].

Reply:
That's a good addition.
But an everyday case can be more involved than just adding it to allow.txt, that's why these instructions are important. Whoever changes any spelling configuration file should read SpellCheckingConfiguration.md at least once to be aware of the options, otherwise you can end up with a ton of garbage dictionary entries (which happened to us).

Reply:
If the dictionary needs to extended, people are to be instructed to read that. This was my intent with the above text, but feel free to adjust it as necessary.

Reply:
Added your quick tips to advice.md, which already had a CTA to read SpellCheckingConfiguration.md.